SSA/Experimental/Bits/Fast/BitStream.lean

```diff
@@ -418,8 +418,37 @@ theorem not_congr (e1 : a ≈ʷ b) : (~~~a) ≈ʷ ~~~b := by
   intros g h
   simp only [not_eq, e1 g h]
 
+theorem ofBitVec_not_eq_to : ofBitVec (~~~ x) ≈ʷ ~~~ ofBitVec x := by
+  intros _ a
+  simp [ofBitVec, a]
+
+theorem negAux_eq_not_addAux : a.negAux = Prod.swap ∘ (~~~a).addAux 1 := by
+  funext i
+  induction' i with _ ih
+  · simp [negAux, addAux, BitVec.adcb, OfNat.ofNat, ofNat]
+  · simp [negAux, addAux, BitVec.adcb, OfNat.ofNat, ofNat, ih]
+
+theorem neg_eq_not_add : - a = ~~~ a + 1 := by
+  ext _
+  simp [negAux_eq_not_addAux, Neg.neg, neg, HAdd.hAdd, Add.add, add, addAux, BitVec.adcb]
+
+theorem ofNat_one (i : Nat) : ofNat 1 i = decide (0 = i) := by
+  cases i
+  <;> simp [ofNat, Nat.shiftRight]
+
+theorem ofBitVec_ofNat : @ofBitVec w 1 ≈ʷ ofNat 1 := by
```

Comment:
```suggestion
theorem ofBitVec_one_eqTo_ofNat : @ofBitVec w 1 ≈ʷ ofNat 1 := by
```
Or `ofBitVec_ofNat_one`, depending on which `ofNat` you intended to refer to in the name (the implicit one for the literal, or the explicit one on the rhs)